src/components/dialog/dialog.js

```diff
@@ -384,7 +384,7 @@ function MdDialogProvider($$interimElementProvider) {
       isolateScope: true,
       onShow: onShow,
       onRemove: onRemove,
-      clickOutsideToClose: true,
+      clickOutsideToClose: false,
```

Comment:
Dialogs shouldn't be that easy to dismiss–users are supposed to use the buttons inside of a modal dialog to either cancel or proceed. Although, this change could cause a negative impact on our existing users, so we could change it back. It's also arguable that clicking outside to close is the same as hitting cancel. I just figured by defaulting `clickOutsideToClose` to false, it would make our dialogs behave more like desktop ones but still allow the option to override.